cmd/hivechain/generate.go

```diff
@@ -86,23 +87,22 @@ func (cfg generatorConfig) addTxForKnownAccounts(i int, gen *core.BlockGen) {
 	}
 
 	txType := (i / cfg.txInterval) % txTypeMax
-	for addr, key := range knownAccounts {
-		if a, ok := cfg.genesis.Alloc[addr]; ok {
-			// It exists, check remaining balance. Would be nice if BlockGen had a way to
-			// check balance, but it doesn't, so we need to estimate the remaining
-			// balance.
+	txCount := 0
+	for txCount <= cfg.txCount {
+		for addr, key := range knownAccounts {
+			// skip account if no allocation
+			if _, ok := cfg.genesis.Alloc[addr]; !ok {
+				continue
+			}
+			log.Printf("TX FROM ACCOUNT: %v", addr)
 			txCost := big.NewInt(1)
-			spent := new(big.Int).Mul(txCost, big.NewInt(int64(i*cfg.txInterval)))
-			gbal := new(big.Int).Set(a.Balance)
-			if gbal.Sub(gbal, spent).Cmp(txCost) < 0 {
-				continue // no funds left in this account
+			if gen.GetBalance(addr).Cmp(txCost) > 0 {
+				tx := generateTx(txType, key, &cfg.genesis, gen)
```

Comment:
You can do it better by always creating the tx, then comparing against `tx.Cost()`